src/puppetlabs/trapperkeeper/services/webserver/jetty9_core.clj

```diff
@@ -561,7 +561,8 @@
   [{:keys [server] :as webserver-context} :- ServerContext]
   (when-let [mbean-container (:mbean-container @(:state webserver-context))]
     (log/debug "Cleaning up JMX MBean container")
-    (.destroy mbean-container))
+    (.destroy mbean-container)
+    (swap! (:state webserver-context) assoc :mbean-container nil))
```

Comment:
Yeah, I agree if (when) we can get back to the world where multiple stops can't be running from different threads things will be much better.  And in that case, you're right that we wouldn't strictly need to worry about making this function re-entrant.  So I'm good with what you have here.